iron-swipeable-container.html

```diff
@@ -0,0 +1,261 @@
+       * should happen over. For example, if the `widthRatio` is 3, the
+       * animation will take place on a distance 3 times the width of the
+       * element being swiped.
+       */
+      widthRatio: {
+       type: Number,
+       value: 3
+      },
+
+      /**
+       * The ratio of the total animation distance after which the opacity
+       * transformation begins. For example, if the `widthRatio` is 1 and
+       * the `opacityRate` is 0.5, then the element needs to travel half its
+       * width before its opacity starts decreasing.
+       */
+      opacityRate: {
+       type: Number,
+       value: 0.2
+      },
+
+      /**
+       * The CSS transition applied while swiping.
+       */
+      transition: {
+        type: String,
+        value: '300ms cubic-bezier(0.4, 0.0, 0.2, 1)'
+      }
+    },
+
+    ready: function() {
+      this._transitionProperty = 'opacity, transform';
+      this._swipeComplete = false;
+      this._direction = '';
+    },
+
+    attached: function() {
+      this._nodeObserver = Polymer.dom(this.$.content).observeNodes(
+          function(mutations) {
+            for (var i = 0; i < mutations.addedNodes.length; i++) {
+              this._addListeners(mutations.addedNodes[i]);
+            }
+            for (var i = 0; i < mutations.removedNodes.length; i++) {
+              this._removeListeners(mutations.removedNodes[i]);
+            }
+          }.bind(this));
+    },
+
+    _addListeners: function(node) {
+      if (node.nodeType === Node.TEXT_NODE)
+        return;
+      // Set up the animation.
+      node.style.transitionProperty = this._transitionProperty;
+      node.style.transition = this.transition;
+
+      this.listen(node, 'track', '_onTrack');
+      this.listen(node, 'transitionend', '_onTransitionEnd');
+    },
+
+    _removeListeners: function(node) {
+      if (node.nodeType === Node.TEXT_NODE)
+        return;
+      this.unlisten(node, 'track', '_onTrack');
+      this.unlisten(node, 'transitionend', '_onTransitionEnd');
+    },
+
+    detached: function() {
+      if (this._nodeObserver) {
+        Polymer.dom(this.$.content).unobserveNodes(this._nodeObserver);
+        this._nodeObserver = null;
+      }
+    },
+
+    _onTrack: function(event) {
+      if (this.disabled)
+        return;
+
+      var target = event.currentTarget;
+      if (target.classList.contains('disable-swipe'))
+        return;
+
+      var track = event.detail;
+      if (track.state === 'start') {
+        this._trackStart(track, target);
+      } else if (track.state === 'track') {
+        this._trackMove(track, target);
+      } else if (track.state === 'end') {
+        this._trackEnd(track, target);
+      }
+    },
+
+    _trackStart: function(event, target) {
+      // Save the width of the element, so that we don't trigger a style
+      // recalc every time we need it.
+      this._nodeWidth = target.offsetWidth;
+      target.style.transition = 'none';
```

Comment:
when the swipe starts, we should disable text selection on the target. Consider adding a class to the `iron-swipeable-container`: `this.toggleClass('swiping', true)`
```css
.swiping {
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  user-select: none;
  cursor: default;
}
```
also, remove the class on swipe end too